src/waveform/renderers/allshader/digitsrenderer.cpp

```diff
@@ -0,0 +1,176 @@
+#include "waveform/renderers/allshader/digitsrenderer.h"
+
+#include <QColor>
+#include <QFontMetricsF>
+#include <QGraphicsBlurEffect>
+#include <QGraphicsPixmapItem>
+#include <QGraphicsScene>
+#include <QOpenGLTexture>
+#include <QPainter>
+#include <QPainterPath>
+
+#include "util/texture.h"
+#include "waveform/renderers/allshader/matrixforwidgetgeometry.h"
+#include "waveform/renderers/allshader/vertexdata.h"
+
+// Render digits using a texture (generated) with digits with blurred dark outline
+
+namespace {
+int charToIndex(char ch) {
+    if (ch >= '0' && ch <= '9') {
+        return static_cast<int>(ch - '0');
+    }
+    if (ch == ':') {
```

Comment:
This and the condition below require a source code comment.

Reply:
Maybe you should also add a indexToChar() to make the code understandable.

Reply:
done!

Reply:
Can I resolve this, @daschuer  ?